weblab/repocache/models.py

```diff
@@ -328,8 +328,7 @@ class ProtocolInterface(models.Model):
 
     A blank term is added to indicate that the interface has been analysed, in case of no actual terms being found.
     """
-    protocol_version = models.ForeignKey(CachedEntityVersion, on_delete=models.SET_NULL, related_name='interface', null=True, db_constraint=False)
-    new_protocol_version = models.ForeignKey(CachedProtocolVersion, on_delete=models.SET_NULL, related_name='interface', null=True, db_constraint=False, default=None)
+    protocol_version = models.ForeignKey(CachedProtocolVersion, on_delete=models.CASCADE, related_name='interface')
```

Comment:
I could do with understanding why the changes have been made for on_delet and for removing db_constraint = false please.

Reply:
I think you're looking just at an intermediate commit there, right? So to get the migrations to operate without losing data, I had to split them into stages, because the `protocol_version` link was moving to point at a different table. So I had stage 1 adding a new field linking to the new table, making both links optional (`null=True`) and ensuring that if either source field was deleted the `ProtocolInterface` entry wouldn't be removed (`SET_NULL`). Then stage 2 gets rid of the old link, and stage 3 renames the new link to have the original name, making it required, and cascading deletions again. The `db_constraint` stuff was to stop the DB complaining during migrations IIRC.